packages/logs/src/domain/assembly.ts

```diff
@@ -35,7 +37,7 @@ export function startLogsAssembly(
       const log = combine(
         {
           service: configuration.service,
-          session_id: session.id,
+          session_id: session.isActiveAt(startTime) ? session.id : undefined,
```

Comment:
**💬 suggestion:** ‏I think `isActiveAt` complicates things quite a bit. It might be simpler to just call `findTrackedSession` multiple times:

```ts
const session = sessionManager.findTrackedSession(startTime)

if (
  !session &&
  (!configuration.sendLogsAfterSessionExpiration || !sessionManager.findTrackedSession(startTime, { returnInactive: true }))) {
  return
}

// and then

session_id: session?.id
```

Reply:
Interesting! It removes the needs to add `endTime` to `SessionContext`. I like it 👍